ygot/struct_validation_map_test.go

```diff
@@ -1822,6 +1908,22 @@ var mergeStructTests = []struct {
 		UnionField: &copyUnionI{42},
 	},
 	wantErr: "interface field was set in both src and dst and was not equal",
+}, {
+	name: "overwrite merge union: values not equal, and different types",
+	inA: &validatedMergeTest{
+		UnionField: &copyUnionS{"greens-amber"},
+	},
+	inB: &validatedMergeTest{
+		UnionField: &copyUnionI{42},
+	},
+	inOpts: []MergeOpt{
+		&MergeStructOpt{
+			Overwrite: true,
+		},
+	},
+	want: &validatedMergeTest{
+		UnionField: &copyUnionI{42},
+	},
```

Comment:
It looks like we're missing a test case for a struct field that is overwritten - can we add one please?